src/main/java/com/ridemart/mapper/UserMapper.java

```diff
@@ -0,0 +1,20 @@
+package com.ridemart.mapper;
+
+import com.ridemart.dto.UserDto;
+import com.ridemart.dto.UserResponseDto;
+import com.ridemart.entity.User;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+import java.util.List;
+
+@Mapper(componentModel = "spring")
+public interface UserMapper {
+
+    User toEntity(UserDto userDto);
+
+    @Mapping(target = "id", source = "id")
```

Comment:
if the properties have the same name and type in the objects, you dont need the Mapping annotation